addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1357,6 +1364,7 @@ const Wysiwyg = Widget.extend({
         const restoreSelection = preserveCursor(this.odooEditor.document);
 
         const $editable = $(OdooEditorLib.closestElement(range.startContainer, '.o_editable') || this.odooEditor.editable);
+        // I now think this (above) is the condition to change
```

Comment:
Unless I'm mistaken, none of these places are the right one, at least for the Social Media snippet. Let me know if I'm wrong but I just had a look and I noticed that the custom media icons from the Social Media snippet are in fact editable using the the `Replace` button in the sidebar. However, they do not react to _double click_, which I assume is the actual issue this task is intending to fix. This is your hint. 😉 

The Company Team snippet issue is more intricate IMO. I had a look and it seems the images are explicitely contentEditable="true", which is technically wrong but is not a big deal, but this is lost when the image is replaced. I suppose the media dialog could add the new magic class to make the image editable by double clicking it. However, one problem would remain in that the icon could not be selected anymore with a single click. To be honest, I'm not sure what should be done here. I have a few options in mind that might be interesting to investigate:

- Replace one of the pictures from the Company Team snippet by another picture (not an icon!). Its `contentEditable=true` attribute is lost. However, there looks to be some js code (probably in website) that adds `contentEditable=true` back to the image when clicking on them, which makes them selectable again. If you can find this code it might be easy to make it target icons as well.

- Another solution would be to change the media dialog so that it keeps the `contentEditable` attribute of the original media. I don't really see why changing a media should change its editable status.

Reply:
Thanks for the hint ! I had no idea that double click was supposed to edit and it wasn't mentioned on the task so I didn't realize it was an issue.
That has been fixed for the custom social media icons I believe.
The "replace" button was actually not working either until I patched `replaceMedia`, which needs review (I'll tag you there).
I'll try again to reproduce the issue with the company team snippet.

Reply:
During my testing of the company team snippet, I did not see any of the images become `contentEditable=True`.
Furthermore, simply by adding the `o_editable_media` in their XML template definition, they are editable from the start and remain so after changing their image, icon, videos. Adding extra columns keeps the added class, as it appears to copy existing columns.
The only "issue" during my testing was with the double-click edit on default-sized icons, but that's only because the icon is too small that its container's edges capture the click. Using the "replace" button, or changing the icon size, or zooming to 300% and landing a pixel perfect double-click all work.

Reply:
Actually, icons do have an issue: by default they have "centered" as alignment but they go left (going from video/image => icon) and only go back to the center when manually setting `center` as the alignment option (even though it's already the selected option, just to be clear).

Reply:
> Actually, icons do have another issue: by default they have "centered" as alignment but they go left (going from video/image => icon) and only go back to the center when manually setting `center` as the alignment option (even though it's already the selected option, just to be clear). Who's that feedback for ? Website ?

Since the change is probably in the media dialog when replacing stuff than in the sidebar options proper, we can do it. You can open a task in RD Spec green in our project for that 👍